app/services/parse_csv_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'logger'
+
+module Services
+  class ParseCsvService
+    def initialize(csv_data)
+      @csv_data = csv_data
+    end
+
+    def call
+      rows = CSV.parse(@csv_data, headers: false)
+      rows.map do |row|
+        validate_row(row)
+      end.compact
+    rescue CSV::MalformedCSVError => e
+      raise "Error parsing CSV: #{e.message}"
+    end
+
+    private
+
+    def validate_row(row)
+      name, y, x = row
+
+      if name.nil? || y.nil? || x.nil?
+        Logger.new($stdout).warn("Malformed row in CSV: #{row}")
```

Comment:
This log message show up when running the tests also, and there are not very usefull to the end user. What will be an alternative to make them more usefull to the user and have cleaner test logs?

Reply:
We could throw an error, but I thought that it's better in this case not to do it and parse the lines that are ok. And log the ones with problems for developers. We can have an array and store the malformed rows and display to the end user to as a warning.

Reply:
Yes  I believe surfacing this warnings to the user via the web response will be the best way to handle this.